### ISSUE TITLE
Props removed when saving

### ISSUE THREAD

**Issue by frederichoule** [sveltejs/language-tools#2594]
### Describe the bug

When VSCode adds an import automatically, it adds it under the `let { data }: { data: PageData } = $props();`, which then gets removed when saving, probably due to auto-formatting.

### Reproduction

1. Create a +page.svelte file
2. Write this: (Normally you wouldn't add the "error" import yourself, it would be added by VSCode when adding that function in your code somewhere in the file.)
```
import type { PageData } from './$types';
let { data }: { data: PageData } = $props();
import { error } from '@sveltejs/kit';
```
3. Save (or format file)
4. This will result in the `let {data}` being deleted.
```
import { error } from '@sveltejs/kit';
import type { PageData } from './$types';
```

### Expected behaviour

Either the `import { error }` should be added before the `let { data }` initially, or the formatting tool places it at the correct place and do not delete the `let { data }`.

### System Info

- OS: macOS 15.1
- IDE: VSCode 1.95.3

### Which package is the issue about?

Svelte for VS Code extension

### Additional Information, eg. Screenshots

![CleanShot 2024-11-18 at 07 32 57@2x](https://github.com/user-attachments/assets/88557f84-97a8-4a14-b4f3-79fac488da3a)
![CleanShot 2024-11-18 at 07 33 06@2x](https://github.com/user-attachments/assets/79366c39-c95d-42f5-b759-46d98935d491)

**Comment by jasonlyu123** [comment]
You probably have the `editor.codeActionsOnSave` config. What is the config? if it has `source.organizeImports` then this is a duplicate of #1590. It's a limitation of how the code is transformed under the hood and we haven't found a way to workaround it yet.

I can't reproduce auto-import adding import to after props. Can you provide a code snippet and instructions for reproducing it?

**Comment by frederichoule** [comment]
Yeah it seems like a duplicate of https://github.com/sveltejs/language-tools/issues/1590

Here is a video that shows when an import is added after the `let { data }`.

https://github.com/user-attachments/assets/3fe88cc8-6bbd-4e47-acb3-ec5ee2a9cc5c

**Comment by frederichoule** [comment]
And I forgot to confirm:

```
    "editor.codeActionsOnSave": {
        "source.organizeImports": "always"
    },
```

**Comment by dummdidumm** [comment]
Fixed the auto import, for the more general problem I defer to #1590

**Comment by dummdidumm** [comment]
Should be fixed in the latest release

**Comment by lts20050703** [comment]
@dummdidumm I can still reproduce it in 109.3.2

**Comment by dummdidumm** [comment]
Please provide a complete code snippet that reproduces this

**Comment by frederichoule** [comment]
> @dummdidumm I can still reproduce [mcullifer's issue](https://github.com/sveltejs/language-tools/issues/2594#issuecomment-2494393136) in 109.3.2

The issues seems to be fixed for me with 109.3.2.

**Comment by lts20050703** [comment]
@dummdidumm 
```svelte
<script lang="ts">
	import {} from "svelte"
	let {}: {} = $props()
</script>
```

102.3.2 formats to
```svelte
<script lang="ts">
	import { } from "svelte"
 {} = $props()
</script>
```

`.vscode/settings.json`
```json
{
	"editor.codeActionsOnSave": { "source.organizeImports": "always" }
}
```

**Comment by frederichoule** [comment]
Also:

```
	let { data }: { data: PageData } = $props();

	import {} from "svelte"
```

becomes:

```
	let { data }:import { } from "svelte";
import {} from "svelte"
```
